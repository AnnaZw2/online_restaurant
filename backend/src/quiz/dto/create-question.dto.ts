import { ApiProperty } from '@nestjs/swagger';
import { IsString, IsNotEmpty } from 'class-validator';
export class CreateQuestionDto {
  @ApiProperty({
    description: 'Question',
    type: String,
    required: true,
    example: 'What is the capital of France?',
  })
  @IsString()
  @IsNotEmpty({ message: 'Question is required' })
  question: string;

  @ApiProperty({
    description: 'Correct Answer',
    type: String,
    required: true,
    example: 'Paris',
  })
  @IsString()
  @IsNotEmpty({ message: 'Field  is required' })
  correctAnswer: string;

  @ApiProperty({
    description: 'Option1 for answer',
    type: String,
    required: true,
    example: 'Warsaw',
  })
  @IsString()
  @IsNotEmpty({ message: 'Field  is required' })
  option1: string;

  @ApiProperty({
    description: 'Option2 for answer',
    type: String,
    required: true,
    example: 'London',
  })
  @IsString()
  @IsNotEmpty({ message: 'Field  is required' })
  option2: string;

  @ApiProperty({
    description: 'Option3 for answer',
    type: String,
    required: true,
    example: 'Berlin',
  })
  @IsString()
  @IsNotEmpty({ message: 'Field  is required' })
  option3: string;
}
