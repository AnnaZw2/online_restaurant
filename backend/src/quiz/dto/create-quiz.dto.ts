import { ApiProperty } from '@nestjs/swagger';
import { IsString, IsNotEmpty, IsNumber, IsDate } from 'class-validator';
export class CreateQuizDto {
  @ApiProperty({
    description: 'Title',
    type: String,
    required: true,
    example: 'Capitals',
  })
  @IsString()
  @IsNotEmpty({ message: 'Title is required' })
  title: string;

  @ApiProperty({
    description: 'Description',
    type: String,
    required: true,
    example: 'Quiz about capitals in Europe',
  })
  @IsString()
  @IsNotEmpty({ message: 'Description is required' })
  description: string;

  @ApiProperty({
    description: 'Likes',
    type: Number,
    example: 0,
    default: 0,
  })
  @IsNumber()
  likes: number = 0;

  @ApiProperty({
    description: 'Creation date',
    type: Date,
    example: new Date(),
    default: new Date(),
  })
  @IsDate()
  creationDate: Date;

  constructor() {
    if (!this.creationDate) {
      this.creationDate = new Date();
    }
  }
}
