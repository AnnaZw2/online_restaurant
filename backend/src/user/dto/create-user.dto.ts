import { ApiProperty } from '@nestjs/swagger';
import { IsString, IsEmail, IsNotEmpty } from 'class-validator';
export class CreateUserDto {
  @ApiProperty({
    description: 'Username',
    type: String,
    required: true,
    example: 'John123',
  })
  @IsString()
  @IsNotEmpty({ message: 'Username is required' })
  username: string;

  @ApiProperty({
    description: 'Email',
    type: IsEmail,
    required: true,
    example: 'john123@mail.com',
  })
  @IsString()
  @IsEmail()
  @IsNotEmpty({ message: 'Email is required' })
  email: string;

  @ApiProperty({
    description: 'Password',
    type: String,
    required: true,
    example: 'myPassword123',
  })
  @IsString()
  password: string;
}
