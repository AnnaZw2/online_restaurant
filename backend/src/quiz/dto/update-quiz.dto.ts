import { PartialType } from '@nestjs/mapped-types';
import { CreateQuizDto } from './create-quiz.dto';
import { ApiProperty } from '@nestjs/swagger';

export class UpdateQuizDto extends PartialType(CreateQuizDto) {
  @ApiProperty({
    title: 'New title',
    description: 'New title for quiz',
    type: String,
    required: false,
  })
  title?: string;

  @ApiProperty({
    title: 'New description',
    description: 'New description for quiz',
    type: String,
    required: false,
  })
  description?: string;

  @ApiProperty({
    title: 'Updated number of likes likes',
    description: 'Updated number of likes for quiz',
    type: Number,
    required: false,
  })
  likes?: number;
}
