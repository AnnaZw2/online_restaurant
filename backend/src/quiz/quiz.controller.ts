import {
  Controller,
  Get,
  Post,
  Body,
  Patch,
  Param,
  Delete,
} from '@nestjs/common';
import { QuizService } from './quiz.service';
import { CreateQuizDto } from './dto/create-quiz.dto';
import { UpdateQuizDto } from './dto/update-quiz.dto';
import {
  ApiBody,
  ApiOperation,
  ApiParam,
  ApiResponse,
  ApiTags,
} from '@nestjs/swagger';
import { ObjectId } from 'mongodb';

@ApiTags('Quizzes')
@Controller('quizzes')
export class QuizController {
  constructor(private readonly quizService: QuizService) {}

  @Post()
  @ApiOperation({
    summary: 'Create quiz',
    description: 'Create quiz and add its id to users array of quizzes',
  })
  @ApiResponse({
    status: 201,
    description: 'The quiz has been successfully created.',
    type: CreateQuizDto,
  })
  create(@Body() createQuizDto: CreateQuizDto) {
    return this.quizService.create(createQuizDto);
  }

  @Get()
  @ApiOperation({
    summary: 'Get all quizzes',
    description: 'Get all quizzes from database',
  })
  @ApiResponse({
    status: 200,
    description: 'All quizzes',
    type: [CreateQuizDto],
  })
  findAll() {
    console.log('find all');
    console.log(this.quizService.findAll());
    return this.quizService.findAll();
  }

  @Get(':id')
  @ApiOperation({
    summary: 'Get quiz by id',
    description: 'Get quiz by id from database',
  })
  @ApiResponse({
    status: 200,
    description: 'The found quiz',
    type: CreateQuizDto,
  })
  @ApiResponse({ status: 404, description: 'Quiz not found' })
  @ApiResponse({ status: 400, description: 'Bad request (eg. invalid id)' })
  findOne(@Param('id') id: string) {
    return this.quizService.findOne(+id);
  }

  @Get(':id/questions')
  @ApiOperation({
    summary: 'Get all questions and answers for quiz',
    description: 'Get all questions and answers for quiz for specified id',
  })
  @ApiParam({
    name: 'id',
    description: 'Quiz id. Must be a valid ObjectId',
    type: Number,
    required: true,
    example: new ObjectId(),
  })
  @ApiResponse({
    status: 200,
    description: 'All questions and answers for quiz',
    type: [CreateQuizDto],
  })
  getAllQuestionsAndAnswers(@Param('id') id: string) {
    return this.quizService.getAllQuestionsAndAnswers(+id);
  }

  @Post(':id/questions')
  @ApiOperation({
    summary: 'Add question to quiz',
    description: 'Add question to quiz for specified id',
  })
  @ApiParam({
    name: 'id',
    description: 'Quiz id. Must be a valid ObjectId',
    type: Number,
    required: true,
    example: new ObjectId(),
  })
  @ApiResponse({
    status: 200,
    description: 'Question added to quiz',
    type: [CreateQuizDto],
  })
  @ApiBody({
    description: 'Question',
    type: CreateQuizDto,
  })
  addQuestionToQuiz(@Param('id') id: string, @Body() questionDto: any) {
    return this.quizService.addQuestionToQuiz(+id, questionDto);
  }

  @Post(':id/questions-multiple')
  @ApiOperation({
    summary: 'Add multiple questions to quiz',
    description: 'Add multiple questions to quiz for specified id',
  })
  @ApiParam({
    name: 'id',
    description: 'Quiz id. Must be a valid ObjectId',
    type: Number,
    required: true,
    example: new ObjectId(),
  })
  @ApiResponse({
    status: 200,
    description: 'Questions added to quiz',
    type: [CreateQuizDto],
  })
  @ApiBody({
    description: 'Array of questions',
    type: [CreateQuizDto],
  })
  addQuestionsToQuiz(@Param('id') id: string, @Body() questionsDto: any) {
    return this.quizService.addQuestionsToQuiz(+id, questionsDto);
  }

  @Patch(':id')
  @ApiOperation({
    summary: 'Update quiz',
    description: 'Update quiz for specified id',
  })
  @ApiParam({
    name: 'id',
    description: 'Quiz id. Must be a valid ObjectId',
    type: Number,
    required: true,
    example: new ObjectId(),
  })
  @ApiResponse({
    status: 200,
    description: 'Quiz updated',
    type: [CreateQuizDto],
  })
  @ApiBody({
    description: 'Quiz',
    type: CreateQuizDto,
  })
  update(@Param('id') id: string, @Body() updateQuizDto: UpdateQuizDto) {
    return this.quizService.update(+id, updateQuizDto);
  }

  @Delete(':id')
  @ApiOperation({
    summary: 'Delete quiz',
    description: 'Delete quiz for specified id',
  })
  @ApiParam({
    name: 'id',
    description: 'Quiz id. Must be a valid ObjectId',
    type: Number,
    required: true,
    example: new ObjectId(),
  })
  @ApiResponse({
    status: 200,
    description: 'Quiz deleted',
    type: [CreateQuizDto],
  })
  remove(@Param('id') id: string) {
    return this.quizService.remove(+id);
  }
}
