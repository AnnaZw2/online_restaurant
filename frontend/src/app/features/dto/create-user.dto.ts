import { CreateQuizDto } from "./create-quiz.dto";

export interface CreateUserDTO {
   readonly username: string;
   readonly email: string;
   readonly password: string;
   readonly quizzes?: CreateQuizDto[];
  }