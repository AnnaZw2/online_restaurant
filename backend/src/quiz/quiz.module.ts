import { Module } from '@nestjs/common';
import { QuizService } from './quiz.service';
import { QuizController } from './quiz.controller';
import { TypeOrmModule } from '@nestjs/typeorm';
import { Quiz } from './entities/quiz.entity';
import { Question } from './entities/question.entity';

@Module({
  imports: [TypeOrmModule.forFeature([Quiz, Question])],
  controllers: [QuizController],
  providers: [QuizService],
})
export class QuizModule {}
