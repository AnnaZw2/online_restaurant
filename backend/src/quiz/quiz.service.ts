import {
  BadRequestException,
  Injectable,
  NotFoundException,
} from '@nestjs/common';
import { CreateQuizDto } from './dto/create-quiz.dto';
import { UpdateQuizDto } from './dto/update-quiz.dto';
import { Quiz } from './entities/quiz.entity';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { Question } from './entities/question.entity';
import { CreateQuestionDto } from './dto/create-question.dto';
@Injectable()
export class QuizService {
  constructor(
    @InjectRepository(Quiz)
    private readonly quizRepository: Repository<Quiz>,
    @InjectRepository(Question)
    private readonly questionRepository: Repository<Question>,
  ) {}

  create(createQuizDto: CreateQuizDto) {
    const newQuiz = this.quizRepository.create(createQuizDto);
    console.log('newQuiz', newQuiz);

    return this.quizRepository.save(newQuiz);
  }

  findAll() {
    return this.quizRepository.find({ relations: ['author'] });
  }

  async findOne(id: number) {
    try {
      const quiz = await this.quizRepository.findOne({
        where: { id },
        relations: ['author'],
      });

      this.validateQuizExists(quiz);
      return quiz;
    } catch (error) {
      if (error instanceof BadRequestException) {
        throw new BadRequestException(error.message);
      } else if (error instanceof NotFoundException) {
        throw new NotFoundException(error.message);
      }
      throw error;
    }
  }
  async update(id: number, updateQuizDto: UpdateQuizDto): Promise<Quiz> {
    // Use query builder to load the quiz entity with its related questions
    const quiz = await this.quizRepository
      .createQueryBuilder('quiz')
      .leftJoinAndSelect('quiz.questions', 'questions')
      .where('quiz.id = :id', { id })
      .getOne();

    if (!quiz) {
      throw new NotFoundException(`Quiz with ID ${id} not found`);
    }

    // Update properties of the quiz entity if they exist in the updateQuizDto
    if (updateQuizDto.title) {
      quiz.title = updateQuizDto.title;
    }
    if (updateQuizDto.description) {
      quiz.description = updateQuizDto.description;
    }
    if (updateQuizDto.likes !== undefined) {
      quiz.likes = updateQuizDto.likes;
    }
    if (updateQuizDto.category) {
      quiz.category = updateQuizDto.category;
    }

    // Save the updated quiz entity
    const updatedQuiz = await this.quizRepository.save(quiz);

    return updatedQuiz;
  }

  async remove(id: number) {
    // Find the quiz
    const quiz = await this.quizRepository.findOne({
      where: { id },
      relations: ['questions'],
    });

    if (!quiz) {
      throw new NotFoundException(`Quiz with ID ${id} not found`);
    }

    // Delete associated questions
    await Promise.all(
      quiz.questions.map(async (question) => {
        await this.questionRepository.delete(question.id);
      }),
    );

    // Delete the quiz
    await this.quizRepository.delete(id);
  }

  async getAllQuestionsAndAnswers(quizId: number) {
    const quiz = await this.quizRepository.findOne({
      where: { id: quizId },
      relations: ['questions'],
    });

    if (!quiz) {
      throw new NotFoundException(`Quiz with ID ${quizId} not found`);
    }

    return quiz.questions;
  }
  //add a single question to a quiz at once
  async addQuestionToQuiz(quizId: number, questionDto: CreateQuestionDto) {
    const quiz = await this.quizRepository.findOneBy({ id: quizId });

    if (!quiz) {
      throw new NotFoundException(`Quiz with ID ${quizId} not found`);
    }

    const newQuestion = this.questionRepository.create({
      ...questionDto,
      quiz,
    });

    await this.questionRepository.save(newQuestion);

    return newQuestion;
  }

  //add multiple questions to a quiz at once by passing an array of questions
  async addQuestionsToQuiz(quizId: number, questionsDto: CreateQuestionDto[]) {
    const quiz = await this.quizRepository.findOneBy({ id: quizId });

    if (!quiz) {
      throw new NotFoundException(`Quiz with ID ${quizId} not found`);
    }

    const newQuestions = questionsDto.map((questionDto) =>
      this.questionRepository.create({ ...questionDto, quiz }),
    );

    await this.questionRepository.save(newQuestions);

    return newQuestions;
  }

  public validateId(id: string): void {
    if (!/^[0-9a-fA-F]{24}$/.test(id)) {
      throw new BadRequestException('Invalid ID format');
    }
  }

  public validateQuizExists(quiz: Quiz): void {
    if (!quiz) {
      throw new NotFoundException('Quiz not found');
    }
  }
}
