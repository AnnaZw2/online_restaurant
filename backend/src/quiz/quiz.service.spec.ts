import { QuizService } from './quiz.service';
import { TestingModule, Test } from '@nestjs/testing';
import { Quiz } from './entities/quiz.entity';
import { Repository } from 'typeorm';
import { ObjectId } from 'mongodb';
import { Question } from './entities/question.entity';
import { getRepositoryToken } from '@nestjs/typeorm';
import { BadRequestException, NotFoundException } from '@nestjs/common';
import { UpdateQuizDto } from './dto/update-quiz.dto';

describe('QuizService', () => {
  let quizRepository: Repository<Quiz>;
  let questionRepository: Repository<Question>;
  let quizService: QuizService;
  const objectId = new ObjectId();

  const mockUser = {
    id: parseInt(objectId.toString(), 16),
    username: 'username',
    email: 'email',
    password: 'password',
    quizzes: [],
  };
  const mockQuiz: Quiz = {
    id: parseInt(objectId.toString(), 16),
    title: 'Quiz title',
    description: 'Quiz description',
    likes: 0,
    creationDate: new Date(),
    author: mockUser as any,
    questions: [],
  };

  const mockQuestion: Question = {
    id: parseInt(objectId.toString()),
    question: 'Question',
    correctAnswer: 'Correct answer',
    option1: 'Option 1',
    option2: 'Option 2',
    option3: 'Option 3',
    quiz: mockQuiz,
  };

  mockQuiz.questions.push(mockQuestion);
  mockUser.quizzes.push(mockQuiz);
  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [
        QuizService,
        {
          provide: getRepositoryToken(Quiz),
          useClass: Repository,
        },
        {
          provide: getRepositoryToken(Question),
          useClass: Repository,
        },
      ],
    }).compile();

    quizRepository = module.get<Repository<Quiz>>(getRepositoryToken(Quiz));
    questionRepository = module.get<Repository<Question>>(
      getRepositoryToken(Question),
    );
    quizService = module.get<QuizService>(QuizService);
  });
  it('should be defined', () => {
    expect(quizRepository).toBeDefined();
    expect(questionRepository).toBeDefined();
  });

  describe('createQuiz', () => {
    it('should create a new quiz', async () => {
      quizRepository.create = jest.fn().mockReturnValue(mockQuiz);
      quizRepository.save = jest.fn().mockResolvedValue(mockQuiz);
      const result = await quizRepository.save(mockQuiz);
      expect(result).toEqual(mockQuiz);
    });
  });

  describe('findAll', () => {
    it('should return an array of quizzes', async () => {
      const quiz = new Quiz();
      quizRepository.find = jest.fn().mockResolvedValue([quiz]);
      const result = await quizRepository.find();
      expect(result).toEqual([quiz]);
    });

    it('should return an empty array', async () => {
      quizRepository.find = jest.fn().mockResolvedValue([]);
      const result = await quizRepository.find();
      expect(result).toEqual([]);
    });
  });

  describe('findOne', () => {
    it('should return a quiz', async () => {
      quizRepository.findOne = jest.fn().mockResolvedValue(mockQuiz);
      const result = await quizRepository.findOne({
        where: { id: mockQuiz.id },
      });
      expect(result).toEqual(mockQuiz);
    });

    it('should throw an error if quiz does not exist', async () => {
      quizRepository.findOne = jest.fn().mockResolvedValue(null);
      await expect(
        quizService.findOne(mockQuiz.id),
      ).rejects.toThrowErrorMatchingSnapshot();
    });
    it('should throw an error for an invalid id format', async () => {
      jest
        .spyOn(quizRepository, 'findOne')
        .mockRejectedValue(new BadRequestException('Invalid ID format'));

      await expect(
        quizService.findOne(123),
      ).rejects.toThrowErrorMatchingSnapshot();
    });
  });

  describe('update', () => {
    it('should update a quiz', async () => {
      const updateQuizDto: UpdateQuizDto = {
        title: 'Updated Title',
        description: 'Updated Description',
      };

      // Mocking the update method to return the updated quiz
      quizRepository.update = jest.fn().mockResolvedValue({
        affected: 1, // Assuming one record was affected
      });

      const result = await quizService.update(mockQuiz.id, updateQuizDto);

      expect(quizRepository.update).toHaveBeenCalledWith(
        mockQuiz.id,
        updateQuizDto,
      );

      expect(result).toEqual({
        affected: 1,
      });
    });
  });

  describe('remove', () => {
    it('should delete a quiz', async () => {
      quizRepository.delete = jest.fn().mockResolvedValue({
        affected: 1,
      });

      const result = await quizService.remove(mockQuiz.id);

      expect(quizRepository.delete).toHaveBeenCalledWith(mockQuiz.id);
      expect(result).toEqual({
        affected: 1,
      });
    });
  });

  describe('getAllQuestionsAndAnswers', () => {
    it('should return questions for a valid quiz ID', async () => {
      // Mocking the findOne method to return a quiz with questions
      quizRepository.findOne = jest.fn().mockResolvedValue({
        id: mockQuiz.id,
        title: mockQuiz.title,
        questions: [
          {
            id: 1,
            question: 'Question 1',
            option1: 'Option 1',
            option2: 'Option 2',
            option3: 'Option 3',
            correctAnswer: 'Correct Answer 1',
          },
        ],
      });

      const result = await quizService.getAllQuestionsAndAnswers(mockQuiz.id);

      // Verify that the findOne method was called with the correct arguments
      expect(quizRepository.findOne).toHaveBeenCalledWith({
        where: { id: mockQuiz.id },
        relations: ['questions'],
      });

      expect(result).toEqual([
        {
          id: 1,
          question: 'Question 1',
          correctAnswer: 'Correct Answer 1',
          option1: 'Option 1',
          option2: 'Option 2',
          option3: 'Option 3',
        },
      ]);
    });

    it('should throw NotFoundException for an invalid quiz ID', async () => {
      quizRepository.findOne = jest.fn().mockResolvedValue(null);

      // Use the expect.assertions to ensure that the assertions in the test are executed
      expect.assertions(1);

      try {
        await quizService.getAllQuestionsAndAnswers(123);
      } catch (error) {
        expect(error).toBeInstanceOf(NotFoundException);
      }
    });
  });

  describe('addQuestionToQuiz', () => {
    it('should add a new question to a valid quiz ID', async () => {
      quizRepository.findOneBy = jest.fn().mockResolvedValue({
        id: mockQuiz.id,
        title: mockQuiz.title,
      });

      questionRepository.create = jest.fn().mockReturnValue({
        id: mockQuestion.id,
        question: 'Question',
        correctAnswer: 'Correct answer',
        option1: 'Option 1',
        option2: 'Option 2',
        option3: 'Option 3',
        quiz: {
          id: mockQuiz.id,
          title: mockQuiz.title,
        },
      });

      questionRepository.save = jest.fn().mockReturnValue({
        id: mockQuestion.id,
        question: 'Question',
        correctAnswer: 'Correct answer',
        option1: 'Option 1',
        option2: 'Option 2',
        option3: 'Option 3',
        quiz: {
          id: mockQuiz.id,
          title: mockQuiz.title,
        },
      });

      const result = await quizService.addQuestionToQuiz(
        mockQuiz.id,
        mockQuestion,
      );

      expect(quizRepository.findOneBy).toHaveBeenCalledWith({
        id: mockQuiz.id,
      });

      expect(questionRepository.create).toHaveBeenCalledWith({
        id: mockQuestion.id,
        question: 'Question',
        correctAnswer: 'Correct answer',
        option1: 'Option 1',
        option2: 'Option 2',
        option3: 'Option 3',
        quiz: {
          id: mockQuiz.id,
          title: mockQuiz.title,
        },
      });
      expect(questionRepository.save).toHaveBeenCalledWith({
        id: mockQuestion.id,
        question: 'Question',
        correctAnswer: 'Correct answer',
        option1: 'Option 1',
        option2: 'Option 2',
        option3: 'Option 3',
        quiz: {
          id: mockQuiz.id,
          title: mockQuiz.title,
        },
      });

      expect(result).toEqual({
        id: mockQuestion.id,
        question: 'Question',
        correctAnswer: 'Correct answer',
        option1: 'Option 1',
        option2: 'Option 2',
        option3: 'Option 3',
        quiz: {
          id: mockQuiz.id,
          title: mockQuiz.title,
        },
      });
    });
    it('should throw NotFoundException for an invalid quiz ID', async () => {
      quizRepository.findOneBy = jest.fn().mockResolvedValue(null);

      expect.assertions(1);

      try {
        await quizService.addQuestionToQuiz(123, mockQuestion);
      } catch (error) {
        expect(error).toBeInstanceOf(NotFoundException);
      }
    });
  });
});
