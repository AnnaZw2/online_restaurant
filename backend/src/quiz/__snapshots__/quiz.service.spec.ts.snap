// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizService findOne should throw an error for an invalid id format 1`] = `"Invalid ID format"`;

exports[`QuizService findOne should throw an error if quiz does not exist 1`] = `"Invalid ID format"`;
